### PR TITLE
Rama inicio

### PR DESCRIPTION
**Pull request by bryanm92s**
Cambios en el diseño del RecyclerView en el FragmentInicio

### CODE REVIEW

**Comment by bryanm92s** [comment]
@jrojas25

File: app/build.gradle [hunk 1]
```diff
@@ -82,6 +82,8 @@ dependencies {
     implementation 'androidx.recyclerview:recyclerview:1.2.1'
 
     // - - Picasso
-    implementation 'androidx.recyclerview:recyclerview:1.1.0'
+    implementation 'androidx.recyclerview:recyclerview:1.2.1'
     implementation 'com.squareup.picasso:picasso:2.71828'
+
+    implementation 'com.github.takusemba:multisnaprecyclerview:2.0.4'
```
Comment (jrojas25) [inline app/build.gradle:88]: the idea of the bootcamp is not to use third-party libraries, so please remove this and do your own implementation

File: app/src/main/java/com/example/apppractica/Fragments/InicioFragment.kt [hunk 2]
```diff
@@ -28,12 +30,8 @@ class InicioFragment : Fragment() {
     private var param1: String? = null
     private var param2: String? = null
```
Comment (jrojas25) [inline app/src/main/java/com/example/apppractica/Fragments/InicioFragment.kt:31]: remove all the unused code

File: app/src/main/java/com/example/apppractica/Fragments/InicioFragment.kt [hunk 3]
```diff
@@ -66,5 +74,25 @@ class InicioFragment : Fragment() {
...
+            "Gingerbread",
+            "Honeycomb",
+            "Ice Cream Sandwich",
+            "Jelly Bean",
+            "KitKat",
+            "Lollipop",
+            "Marshmallow",
+            "Nougat",
+            "Oreo"
+        )*/
```
Comment (jrojas25) [inline app/src/main/java/com/example/apppractica/Fragments/InicioFragment.kt:95]: remove the unused code

File: app/src/main/java/com/example/apppractica/adapter/BestMoviesAdapter.kt [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+import kotlinx.android.synthetic.main.item_horizontal.view.*
+import kotlinx.android.synthetic.main.layout_item_view.view.*
+
+
+class BestMoviesAdapter(private val bestMovieList: List<BestMovies>) : androidx.recyclerview.widget.RecyclerView.Adapter<BestMoviesAdapter.ViewHolder>() {
+
+
+  override fun onCreateViewHolder(viewGroup: ViewGroup, viewType: Int): ViewHolder {
+    val inflater = LayoutInflater.from(viewGroup.context)
+    val view = inflater.inflate(R.layout.item_horizontal, viewGroup, false)
```
Comment (jrojas25) [inline app/src/main/java/com/example/apppractica/adapter/BestMoviesAdapter.kt:21]: please use viewbinding

File: app/src/main/java/com/example/apppractica/adapter/BestMoviesAdapter.kt [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+
+    fun render(items: BestMovies) {
+
+      Picasso.get().load(items.logo).into(itemView.ivLogo_)
+      stars.text= items.stars.toString()
+      title.text = items.title
+
+      /*itemView.ivLogo.setOnClickListener {
+          Toast.makeText(itemView.ivLogo.context, items.title, Toast.LENGTH_SHORT).show()
+      }*/
```
Comment (jrojas25) [inline app/src/main/java/com/example/apppractica/adapter/BestMoviesAdapter.kt:54]: remove the unused code

File: app/src/main/java/com/example/apppractica/adapter/BestMoviesAdapter.kt [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+        Toast.makeText(itemView.ivLogo_.context,items.title, Toast.LENGTH_SHORT).show()
+
+        //val context=itemView.ivLogo.context
+
+        //val intent = Intent(itemView.ivLogo.context, MovieDetail::class.java)
+
+        //intent.putExtra("Title", items.title)
+        //intent.putExtra("Cast", items.cast)
+        //itemView.context.startActivity(intent);
+        //itemView.ivLogo.context.startActivity(intent)
```
Comment (jrojas25) [inline app/src/main/java/com/example/apppractica/adapter/BestMoviesAdapter.kt:68]: remove the unused code

File: app/src/main/java/com/example/apppractica/adapter/MovieAdapter.kt [hunk 7]
```diff
@@ -3,23 +3,29 @@ package com.example.apppractica.adapter
...
+class MovieAdapter(private val movieList: List<Movies>) : androidx.recyclerview.widget.RecyclerView.Adapter<MovieAdapter.ViewHolder>() {
 
-    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MovieViewHolder {
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
         //lateinit var movieHolder: MovieHolder
         //movieHolder = MovieHolder(LayoutInflater.from(parent.context).inflate(R.layout.layout_item_view,parent, false))
         //return movieHolder
         val layoutInflater = LayoutInflater.from(parent.context)
-        return MovieViewHolder(layoutInflater.inflate(R.layout.layout_item_view,parent, false))
+        return ViewHolder(layoutInflater.inflate(R.layout.layout_item_view,parent, false))
```
Comment (jrojas25) [inline app/src/main/java/com/example/apppractica/adapter/MovieAdapter.kt:24]: use viewBinding

File: app/src/main/java/com/example/apppractica/adapter/MovieAdapter.kt [hunk 8]
```diff
@@ -33,16 +39,45 @@ class MovieAdapter(private val movieList: List<Movies>) : RecyclerView.Adapter<M
...
+            MovieCast.text=items.cast
 
-    /*class MovieHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
-        //var movieLogo = itemView.movielogo
-        //var movieTitle = itemView.movietitle
-        //var movieDate = itemView.moviedate
-        //var movieCast = itemView.moviecast
+            /*itemView.ivLogo.setOnClickListener {
+                Toast.makeText(itemView.ivLogo.context, items.title, Toast.LENGTH_SHORT).show()
+            }*/
```
Comment (jrojas25) [inline app/src/main/java/com/example/apppractica/adapter/MovieAdapter.kt:61]: remove the unused code

File: app/src/main/java/com/example/apppractica/adapter/MovieAdapter.kt [hunk 9]
```diff
@@ -33,16 +39,45 @@ class MovieAdapter(private val movieList: List<Movies>) : RecyclerView.Adapter<M
...
+                Toast.makeText(itemView.ivLogo.context,items.title, Toast.LENGTH_SHORT).show()
 
+                //val context=itemView.ivLogo.context
+
+                //val intent = Intent(itemView.ivLogo.context, MovieDetail::class.java)
+
+                //intent.putExtra("Title", items.title)
+                //intent.putExtra("Cast", items.cast)
+                //itemView.context.startActivity(intent);
+                //itemView.ivLogo.context.startActivity(intent)
```
Comment (jrojas25) [inline app/src/main/java/com/example/apppractica/adapter/MovieAdapter.kt:75]: remove the unused code

File: app/src/main/java/com/example/apppractica/adapter/MovieAdapter.kt [hunk 10]
```diff
@@ -33,16 +39,45 @@ class MovieAdapter(private val movieList: List<Movies>) : RecyclerView.Adapter<M
...
-        //var movieCast = itemView.moviecast
+            /*itemView.ivLogo.setOnClickListener {
+                Toast.makeText(itemView.ivLogo.context, items.title, Toast.LENGTH_SHORT).show()
+            }*/
 
-    }*/
+            // Gestionar clicks en el cuadro del RecyclerView
 
+            itemView.setOnClickListener {
+                Toast.makeText(itemView.ivLogo.context,items.title, Toast.LENGTH_SHORT).show()
```
Comment (jrojas25) [inline app/src/main/java/com/example/apppractica/adapter/MovieAdapter.kt:66]: itemView is a view itself, so you don't need ivLogo to get the context

File: app/src/main/res/layout/fragment_buscar.xml [hunk 11]
```diff
@@ -1,14 +1,41 @@
...
     android:layout_height="match_parent"
     tools:context=".Fragments.BuscarFragment">
 
-    <androidx.recyclerview.widget.RecyclerView
-        android:id="@+id/movieRecycler"
+    <LinearLayout
         android:layout_width="match_parent"
         android:layout_height="wrap_content"
-        />
+        android:layout_margin="10dp"
```
Comment (jrojas25) [inline app/src/main/res/layout/fragment_buscar.xml:11]: use dimens

File: app/src/main/res/layout/fragment_buscar.xml [hunk 12]
```diff
@@ -1,14 +1,41 @@
...
\ No newline at end of file
+        <RelativeLayout
+            android:layout_width="match_parent"
+            android:layout_height="?actionBarSize">
+
+            <ImageView
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:layout_centerVertical="true"
+                android:padding="15dp"
```
Comment (jrojas25) [inline app/src/main/res/layout/fragment_buscar.xml:22]: use dimens

File: app/src/main/res/layout/fragment_inicio.xml [hunk 13]
```diff
@@ -1,14 +1,77 @@
...
+                    android:orientation="vertical">
+
+                    <TextView
+                        android:layout_width="wrap_content"
+                        android:layout_height="wrap_content"
+                        android:layout_margin="20dp"
+                        android:fontFamily="@font/robotobold"
+                        android:textColor="@color/color_negro"
+                        android:drawableLeft="@drawable/ic_baseline_stars_24"
+                        android:text="Las mejores selecciones"
```
Comment (jrojas25) [inline app/src/main/res/layout/fragment_inicio.xml:60]: use the strings file

**Review by jrojas25 — COMMENTED** [review]
some of the comments apply to other classes/files, so please review each one to fulfill the recommendations

**Pull request merged**